src/jitlog/reader.lua

```diff
@@ -26,6 +32,34 @@ function base_actions:stringmarker(msg)
   return marker
 end
 
+function base_actions:gcstring(msg)
+  local string = msg:get_data()
+  local address = addrtonum(msg.address)
+  self.strings[address] = string
+  self:log_msg("gcstring", "GCstring: %s, %s", address, string)
+  return string, address
+end
+
+function base_actions:gcproto(msg)
+  local address = addrtonum(msg.address)
+  local chunk = self.strings[addrtonum(msg.chunkname)]
+  local proto = {
+    chunk = chunk, 
+    firstline = msg.firstline, 
+    numline = msg.numline,
+    bclen = msg.bclen,
+    bcaddr = addrtonum(msg.bcaddr),
+    address = address,
+  }
+  proto.hotslot = band(rshift(proto.bcaddr, 2), 64-1)
+  self.protos[#self.protos + 1] = proto
+  self.proto_lookup[address] = proto
+  -- msg:get_bc()
```

Comment:
Dead code?

Reply:
its kind of a bad to do i need to save away the byte code so we can use it for things like finding what bytecode traces are aborting on

Reply:
Our rules is that, except in very unusual circumstances, we don't commit commented code. I'm very happy to have an assert or exception here though.

Reply:
Check the 3 new commits the fields should now be used as well as a expanded proto test to use them